src/test/compile-fail/issue-41394.rs

```diff
@@ -9,8 +9,9 @@
 // except according to those terms.
 
 enum Foo {
-    A = "" + 1
+    A = "" + 1,
     //~^ ERROR binary operation `+` cannot be applied to type `&str`
+    B = 42,
```

Comment:
Why would you need this? Evaluating `Foo::A as isize` should *most definitely* evaluate `Foo::A`'s discriminant expression.

Reply:
I think a previous iteration if this PR failed with an error on `Bar` before it failed on the error on `Foo`, so `Foo::B` isn't necessary anymore